src/map/script.cpp

```diff
@@ -24026,6 +24026,50 @@ BUILDIN_FUNC(achievement_condition){
 	return SCRIPT_CMD_SUCCESS;
 }
 
+/*
+  charinfo(<char_id>,<type>)
+  charinfo(<account_id>,<type>)
+  charinfo(<player_name>,<type>)
+*/
+BUILDIN_FUNC(charinfo) {
+	TBL_PC *sd;
+
+	if (script_isstring(st, 2))
+		sd = map_nick2sd(script_getstr(st, 2),false);
+	else {
+		int id = script_getnum(st, 2);
+		sd = map_id2sd(id);
+		if (sd == NULL)
+			sd = map_charid2sd(id);
+	}
+
+	switch( script_getnum(st, 3) ) {
+	case PC_NAME:
+		if (sd)
```

Comment:
The if-statements could be used outside.
Pseudocode:
`
if(sd) {
    Switch() ...
} else {
   Switch() ...
}`
But the compiler will optimize it that way anyway. So up to you.